include/nbl/builtin/hlsl/math/complex.hlsl

```diff
@@ -113,32 +108,54 @@ struct complex_t
     // ----------------- Relational operators -----------------------------
     bool operator==(const complex_t rhs)
     {
-            return !(uint64_t(m_real) ^ uint64_t(rhs.m_real)) && !(uint64_t(m_imag) ^ uint64_t(rhs.m_imag));
+            return m_real == rhs.m_real && m_imag == rhs.m_imag;
     }
     bool operator!=(const complex_t rhs)
     {
-            return uint64_t(m_real) ^ uint64_t(rhs.m_real) || uint64_t(m_imag) ^ uint64_t(rhs.m_imag);
+            return m_real != rhs.m_real || m_imag != rhs.m_imag;
     }
-            
 };
 
-// ---------------------- Compound assignment functors ----------------------
-template <typename Scalar>
-struct ComplexCompoundHelper{              
-    using add = assign_add_t<complex_t<Scalar> >;
-    using subtract = assign_subtract_t<complex_t<Scalar> >;
-    using mul = assign_mul_t<complex_t<Scalar> >;
-    using div = assign_div_t<complex_t<Scalar> >;
+// ---------------------- Compound assign operators ------------------------
+// Specializations of the structs found in functional.hlsl
+// TODO: Figure out a way of making them have an identity (const static member struct doesn't compile on Godbolt)
```

Comment:
oh yes, open a DXC issue about this!

Reply:
aaah its a template, you're blocked by https://github.com/microsoft/hlsl-specs/issues/109

Reply:
you might have to explicitly out-of-line initialize every specialization combo of `complex_t`